packages/opentelemetry-host-metrics/src/metric.ts

```diff
@@ -164,63 +134,72 @@ export class HostMetrics extends BaseMetrics {
+      enums.METRIC_NAMES.CPU_UTILIZATION,
+      {
         description: 'Cpu usage time 0-1',
-      })
-      .addCallback(observableResult => {
-        const cpuUsageData = this._getCpuUsageData();
-        this._updateCpuUtilisation(observableResult, cpuUsageData);
-      });
-    this._meter
-      .createObservableGauge(enums.METRIC_NAMES.MEMORY_USAGE, {
+      }
+    );
+
+    this._memoryUsage = this._meter.createObservableGauge(
+      enums.METRIC_NAMES.MEMORY_USAGE,
+      {
         description: 'Memory usage in bytes',
-      })
-      .addCallback(observableResult => {
-        const memoryUsageData = this._getMemoryData();
-        this._updateMemUsage(observableResult, memoryUsageData);
-      });
-    this._meter
-      .createObservableGauge(enums.METRIC_NAMES.MEMORY_UTILIZATION, {
+      }
+    );
+    this._memoryUtilization = this._meter.createObservableGauge(
+      enums.METRIC_NAMES.MEMORY_UTILIZATION,
+      {
         description: 'Memory usage 0-1',
-      })
-      .addCallback(observableResult => {
-        const memoryUsageData = this._getMemoryData();
-        this._updateMemUtilization(observableResult, memoryUsageData);
-      });
-    this._meter
-      .createObservableCounter(enums.METRIC_NAMES.NETWORK_DROPPED, {
+      }
+    );
+
+    this._networkDropped = this._meter.createObservableCounter(
+      enums.METRIC_NAMES.NETWORK_DROPPED,
+      {
         description: 'Network dropped packets',
-      })
-      .addCallback(async observableResult => {
-        const networkData = await this._getNetworkData();
-        this._updateNetworkDropped(observableResult, networkData);
-      });
-    this._meter
-      .createObservableCounter(enums.METRIC_NAMES.NETWORK_ERRORS, {
+      }
+    );
+    this._networkErrors = this._meter.createObservableCounter(
+      enums.METRIC_NAMES.NETWORK_ERRORS,
+      {
         description: 'Network errors counter',
-      })
-      .addCallback(async observableResult => {
-        const networkData = await this._getNetworkData();
-        this._updateNetworkErrors(observableResult, networkData);
-      });
-    this._meter
-      .createObservableCounter(enums.METRIC_NAMES.NETWORK_IO, {
+      }
+    );
+    this._networkIo = this._meter.createObservableCounter(
+      enums.METRIC_NAMES.NETWORK_IO,
+      {
         description: 'Network transmit and received bytes',
-      })
-      .addCallback(async observableResult => {
-        const networkData = await this._getNetworkData();
-        this._updateNetworkIO(observableResult, networkData);
-      });
+      }
+    );
+
+    this._meter.addBatchObservableCallback(
+      async observableResult => {
+        const cpuUsages = getCpuUsageData();
+        const memoryUsage = getMemoryData();
+        const networkData = await getNetworkData();
+
+        this._batchUpdateCpuUsages(observableResult, cpuUsages);
+        this._updateMemUsage(observableResult, memoryUsage);
```

Comment:
NIT: I can see you changed the name of updateCpuUsages to include batch at the beginning, do you want to update _updateMemUsage and _updateNetworkData as well?, it will be easier to read the code